src/LinuxTracing/LinuxTracingIntegrationTest.cpp

```diff
@@ -946,24 +960,34 @@ TEST(LinuxTracingIntegrationTest, ThreadNames) {
 
   VerifyOrderOfAllEvents(events);
 
-  std::vector<std::string> collected_event_names;
+  std::vector<std::string> changed_thread_names;
+  std::vector<std::string> initial_thread_names;
   for (const auto& event : events) {
+    if (event.event_case() == orbit_grpc_protos::ProducerCaptureEvent::kCaptureStarted) {
+      const orbit_grpc_protos::CaptureStarted& capture_started = event.capture_started();
+      for (const auto& thread_name : capture_started.thread_names()) {
+        if (thread_name.pid() != fixture.GetPuppetPid()) {
+          continue;
+        }
```

Comment:
As for the comment below, this doesn't test the `tid`, which we can test.

Reply:
added a check